pkg/k8s/dialer.go

```diff
@@ -110,6 +119,63 @@ func detectConnSuccess(connectSuccess chan struct{}) io.Writer {
 	return pw
 }
 
+var (
+	connectionRefusedErrorRE = regexp.MustCompile(`E connect\(\d+, AF=\d+ (?P<hostport>[\[\]0-9.:a-z]+), \d+\): Connection refused`)
+	nameResolutionErrorRE    = regexp.MustCompile(`E getaddrinfo\("(?P<hostname>[a-zA-z-.0-9]+)",.*\): Name does not resolve`)
```

Comment:
I'm just going to have to trust you on these regexes 😆

Reply:
Yeah, wasn't easy to write them. I am glad that https://regex101.com/ exists.